wlib/exception/Exception.h

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file Exception.h
+ * @brief Custom exception types for WLib.
+ *
+ * @author Jeff Niu
+ * @date November 2, 2017
+ * @bug No known bugs
+ */
+
+/*
+ * TODO: Use dynamic string type once implemented
+ * TODO: Define RangeException
+ */
+
+#ifndef EMBEDDEDTESTS_EXCEPTION_H
```

Comment:
We technically cannot throw exceptions on embedded systems especially on Arduino but we can use this custom Exception class and modify it to give Logs and do Assertions. Since it is a custom class seems like a nice thing to implement